dpctl/tensor/libtensor/include/kernels/elementwise_functions/common.hpp

```diff
@@ -615,6 +690,261 @@ typedef sycl::event (*binary_contig_row_contig_matrix_broadcast_impl_fn_ptr_t)(
     py::ssize_t,
     const std::vector<sycl::event> &);
 
+template <typename argTy1,
+          typename argTy2,
+          template <typename T1, typename T2>
+          class BinaryOutputType,
+          template <typename T1,
+                    typename T2,
+                    typename T3,
+                    unsigned int vs,
+                    unsigned int nv>
+          class BinaryContigFunctorT,
+          template <typename T1,
+                    typename T2,
+                    typename T3,
+                    unsigned int vs,
+                    unsigned int nv>
+          class kernel_name>
+sycl::event binary_contig_impl(sycl::queue exec_q,
+                               size_t nelems,
+                               const char *arg1_p,
+                               py::ssize_t arg1_offset,
+                               const char *arg2_p,
+                               py::ssize_t arg2_offset,
+                               char *res_p,
+                               py::ssize_t res_offset,
+                               const std::vector<sycl::event> &depends = {})
+{
+    sycl::event comp_ev = exec_q.submit([&](sycl::handler &cgh) {
+        cgh.depends_on(depends);
+
+        size_t lws = 64;
+        constexpr unsigned int vec_sz = 4;
+        constexpr unsigned int n_vecs = 2;
```

Comment:
These could be moved into the template parameters like the unary functions.

Reply:
Yes, good point. A change incoming